model-optimizer/extensions/ops/ctc_greedy_decoder.py

```diff
@@ -47,4 +47,7 @@ def ctc_greedy_decoder_infer(node: Node):
         outn.shape = np.ones(4, dtype=np.int)
         assert inn.shape[1] == inn2.shape[1], 'Batch for CTCGreedyDecoder should be the same in both inputs'
         outn.shape[0] = inn.shape[1]
-        outn.shape[1] = inn.shape[0]
+        if node.graph.graph['layout'] == 'NHWC':
```

Comment:
Looks like an ugly solution. Also it is not clear why we solve it here that way. There should be at least some comment describing how it works.